core/src/main/clojure/xtdb/expression/metadata.clj

```diff
@@ -23,16 +20,20 @@
 
 (declare meta-expr)
 
+(def ^:private bool-metadata-types #{:null :bool :fixed-size-binary :clj-form})
+
 (defn call-meta-expr [{:keys [f args] :as expr} {:keys [col-types] :as opts}]
   (letfn [(var-param-expr [f meta-value field {:keys [param-type] :as param-expr}]
-            (let [base-col-types (-> (get col-types field)
-                                     types/flatten-union-types)]
-              (simplify-and-or-expr
-                {:op :call
-                 :f :or
-                 ;; TODO this seems like it could make better use
-                 ;; of the polymorphic expr patterns?
-                 :args (vec
+            ;; TODO adapt for boolean metadata writer
```

Comment:
cheers for raising the other card - not sure I entirely understand its impact, shall we discuss tomorrow?